FS19/ML/Zusammenfassung_ML.tex

```diff
@@ -1488,4 +1491,193 @@ \subsubsection{k-fold Cross Validation}
+	\text{support}(X\rightarrow Y) = \text{support}(X\cup Y)
+\end{equation*}
+
+The support is \emph{direction invariant} and therefore cannot measure the quality of a directed rule
+\begin{equation*}
+	\text{support}(X\rightarrow Y) = \text{support}(Y\rightarrow X)
+\end{equation*}
+
+\begin{figure}[tbh!]
+	\centering
+	\includegraphics[width=0.7\linewidth, keepaspectratio]{Pictures/support_association_rule}
+	\caption{Support of association rules}
+	\label{fig:supportassociationrule}
+\end{figure}
+
+The support measures how frequently an item set occurs in the data. Rules with low support may occur simply by chance, while \textbf{good association rules have a high support}.
+
+\begin{theorem}
+	Support = Interestingness
+\end{theorem}
+
+\subsubsection{Confidence of an Association Rule}
+
+The \textbf{confidence} of an association rule is defined as
+
+\begin{equation*}
+	\text{confidence}(X\rightarrow Y) = \frac{\text{support}(X\cup Y)}{\text{support}(X)}
+\end{equation*}
+
+\begin{figure}[tbh!]
+	\centering
+	\includegraphics[width=0.7\linewidth, keepaspectratio]{Pictures/confidence_association_rule}
+	\caption{Examples of confidence}
+	\label{fig:confidenceassociationrule}
+\end{figure}
+
+For a given rule $X \rightarrow Y$, the higher the confidence, the more likely it is for $Y$ to be present in transactions that contain $X$. Confidence therefore measures reliability or trustworthiness of a rule, and a \textbf{good association rule has high confidence}.
+
+\begin{theorem}
+	Confidence = Trustworthiness
+\end{theorem}
+
+\subsection{Apriori Algorithm}
+
+Given a set of transaction we want to find all rules having a minimum support $\geq \text{min}_s$ and a minimum confidence $\geq \text{min}_c$. The Apriori Algorithm finds such association rules in two steps:
+\begin{enumerate}
+	\item Generate frequent item sets satisfying the support threshold
+	\item Extract rules from frequent item sets satisfying the confidence threshold
+\end{enumerate}
+
+The Apriori Algorithm has the following two properties, the second one allows for efficient pruning (refer to figure \ref{fig:supersetpruning}):
+
+\begin{enumerate}
+	\item If an item set is \textbf{frequent} (has high support), all of its subsets must be frequent too
+	\item If an item set is \textbf{infrequent} (has low support), all of its super-sets must be infrequent too
+\end{enumerate}
+
+\begin{figure}[tbh!]
+	\centering
+	\includegraphics[width=0.5\linewidth, keepaspectratio]{Pictures/superset_pruning}
+	\caption{Pruning of supersets of sets with low support}
+	\label{fig:supersetpruning}
+\end{figure}
+
+After the pruning we generate Rules from the Frequent Item Sets and prune this tree as well. The pruning method for pruning is based on the rule that if a rule $[X \rightarrow Y - X]$ violates the confidence threshold, then any rule $[X’ \rightarrow Y - X’]$ with $X’$ being a subset of $X$ violates the confidence threshold as well (refer to figure \ref{fig:subsetpruning})
```

Comment:
the pruning method for pruning? Is that supposed to be like that or is there a pruning too many?

Reply:
We should prune the pruning